### PR TITLE
fix!: simplify no longer removes neg, add to_py

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -175,23 +175,25 @@ def is_number(self) -> bool:
         """
         Checks whether a Literal expression is a number.
         """
-        return isinstance(self, Literal) and not self.args["is_string"]
+        return (
+            isinstance(self, Literal)
+            and not self.args["is_string"]
+            or isinstance(self, Neg)
+            and self.this.is_number
```
Comment (georgesittas) [inline sqlglot/expressions.py:182]: should we wrap `or`'s operands here to make the order easier to see?

**Review by georgesittas — APPROVED** [review]
Nice!

File: sqlglot/dialects/postgres.py [hunk 2]
```diff
@@ -56,12 +56,15 @@ def func(self: Postgres.Generator, expression: DATE_ADD_OR_SUB) -> str:
         this = self.sql(expression, "this")
         unit = expression.args.get("unit")
 
-        expression = self._simplify_unless_literal(expression.expression)
-        if not isinstance(expression, exp.Literal):
+        e = self._simplify_unless_literal(expression.expression)
+        if isinstance(e, exp.Literal):
+            e.args["is_string"] = True
+        elif e.is_number:
+            e = exp.Literal.string(e.to_py())
```
Comment (VaggelisD) [inline sqlglot/dialects/postgres.py:63]: Is the order in the branches correct?

 If `e.is_number` is `True` then `e` is an `exp.Literal(is_string=False)`, I think the `elif` branch won't be hit

Comment (georgesittas) [reply sqlglot/dialects/postgres.py:63]: > If `e.is_number` is `True` then `e` is an `exp.Literal(is_string=False)`

This isn't true, `is_number` now handles `Neg` expressions as well. This condition is supposed to handle `Neg`s.

Comment (VaggelisD) [reply sqlglot/dialects/postgres.py:63]: Sg, missed the new `is_number` definition

**Comment by achicoine-coveo** [comment]
Thanks for the fix! I can confirm it worked and I was able to remove the macro I had as a workaround :)

**Pull request merged**